File: backend/app/models/response/history.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class HistoryItemResponse(BaseModel):
    id: str
    dataset_name: str
    dataset_id: str | None = None
    algorithm: str
    hyperparameters: dict[str, Any]
    status: str | None = None
    top_metric: dict[str, Any] | None = None
    metrics: dict[str, Any] | None = None
    created_at: datetime
    started_at: datetime | None = None
    completed_at: datetime | None = None

    model_config = {"from_attributes": True}


class HistoryListResponse(BaseModel):
    items: list[HistoryItemResponse]
    total: int
    page: int
    limit: int


class CompareResponse(BaseModel):
    experiments: list[HistoryItemResponse]

File: frontend/src/features/history/types/history.ts
import type { Metrics } from '@/features/experiments/types/index';

export interface HistoryExperiment {
  id: string;
  experiment_id: string;
  algorithm: string;
  dataset_name: string;
  metrics: Metrics;
  created_at: string;
}

export interface HistoryListResponse {
  items: HistoryExperiment[];
  total: number;
  page: number;
  size: number;
}

export interface CompareRequest {
  experiment_ids: string[];
}

export interface CompareResult {
  experiments: {
    experiment_id: string;
    algorithm: string;
    dataset_name: string;
    metrics: Metrics;
  }[];
}
